### PR TITLE
added logging to grat data pipeline, refactored

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner I'm getting deserialization exceptions, logging to see why.

### CODE REVIEW

**Comment by andrewconner** [comment]
:+1:

File: kifi-backend/modules/heimdal/conf/heimdalService.routes [hunk 1]
```diff
@@ -28,7 +28,7 @@ POST /internal/heimdal/data/reKeep/getReKeepCountsByKeepIds         @com.keepit.
 
 GET  /internal/heimdal/data/userKeepInfo/getReKeepCountsByUserUri @com.keepit.controllers.UserKeepInfoController.getReKeepCountsByUserUri(userId: Id[User], uriId: Id[NormalizedURI])
 
-POST  /internal/heimdal/data/getEligibleGratData         @com.keepit.controllers.GratificationStatsController.getEligibleGratData()
+POST  /internal/heimdal/data/getEligibleGratDatas         @com.keepit.controllers.GratificationStatsController.getEligibleGratDatas()
```
Comment (andrewconner) [inline kifi-backend/modules/heimdal/conf/heimdalService.routes:31]: Is this updated in `Router.scala`? 

``` scala
    def getEligibleGratData() = ServiceRoute(POST, "/internal/heimdal/data/getEligibleGratData")
```

Comment (camhashemi) [inline kifi-backend/modules/heimdal/conf/heimdalService.routes:31]: Negative. Thank you.

File: kifi-backend/modules/heimdal/app/com/keepit/controllers/GratificationStatsController.scala [hunk 2]
```diff
@@ -22,20 +22,23 @@ class GratificationStatsController @Inject() (
...
     val userIds = request.body.as[Seq[Id[User]]]
     val gratDatas = userIds.map { gratStatsCommander.getGratData }.filter { _.isEligible }
+    log.info(s"[GratData] Eligible Grat Datas collected. Sending: ${Json.toJson(gratDatas)}")
     Ok(Json.toJson(gratDatas))
   }
 
   def getGratData(userId: Id[User]) = Action { request =>
     val gratData = gratStatsCommander.getGratData(userId)
+    log.info(s"[GratData] Grat Data collected. Sending: ${Json.toJson(gratData)}")
     Ok(Json.toJson(gratData))
```
Comment (camhashemi) [inline kifi-backend/modules/heimdal/app/com/keepit/controllers/GratificationStatsController.scala]: @andrewconner so what I'm getting back from this is a JsArray containing the JsArray of GratDatas, causing the deserializer expecting one JsArray to throw an exception. Any idea why this is happening? I figure that `Json.obj("gratDatas" -> Json.toJson(gratDatas))` will work.

**Comment by camhashemi** [comment]
@andrewconner updated

**Pull request merged**